spec/support/rpush_helpers.rb

```diff
@@ -1,5 +1,9 @@
 module RpushHelpers
-  def mock_rpush(consumer_app)
+  # This method creates the mocks necessary to test RPush related features since
+  # we use the fakeredis gem and don't have a working Redis server in CI.
+  # The param `empty` allows us to conditionally mock if the app exists or not
+  # Example: Rpush::Apns2::App.where(...).first => nil/app (empty true/false)
+  def mock_rpush(consumer_app, empty: false)
```

Comment:
This returns a hash with one key-value-pair which means we have to do this:

```ruby
mocked_objects = mock_rpush(device.consumer_app)
expect(mocked_objects[:rpush_notification]).to
```

Is there a reason we can't just return the created `rpush_notification` directly so this code becomes:

```ruby
rpush_notifiation = mock_rpush(device.consumer_app)
expect(rpush_notification).to ...
```

Reply:
> Is there a reason we can't just return the created rpush_notification directly so this code becomes...

I would have to speculate a bit, because this was originally added by @rhymes when he added `fakeredis` support.

The idea behind it is that we're mocking `Rpush::Gcm::App` or `Rpush::Apns2::App` so they are able to create the respective notifications (`Rpush::Gcm::Notification` / `Rpush::Apns2::Notification`). In the specs we don't work with the App models directly, instead we just call our Services (`PushNotifications::Send`) after `mock_rpush` is called and then check against the notifications like you mentioned

```ruby
expect(mocked_objects[:rpush_notification]).to have_received(:save!)....
```

My guess is that it was returning a hash with one key-value-pair because that's the only one needed at the moment, but the pattern of "mocking rpush" could potentially need more objects to be returned in the future.

So the answer would be no, there's no reason we can't just directly return the notification. It might have a bit more semantic value to "mock rpush" and get a bunch of underlying objects that were mocked though, despite being a bit more verbose (specially in this case were we only have one that we need at the moment). 

All of this to say that I'm on the fence here about changing this. Since it's a _non-blocking request/question_ I'd rather wait for a later time to hear from Rhymes if the hash response is preferred for some other reason. Unless anyone else feels we should definitely change this, then I'd be happy to push the update 😄

Reply:
> All of this to say that I'm on the fence here about changing this.

It was purely a clarifying question, not a request to change anything, I should have made that clearer.